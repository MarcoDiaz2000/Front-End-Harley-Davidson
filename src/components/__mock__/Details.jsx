import React from 'react';

const Details = () => (
  <div className="relative w-full h-full">
    <div className="w-full h-full flex justify-center items-center flex-col md:flex-row gap-2 px-4 py-4 font-roboto">
      <div className="flex flex-grow justify-center item-center rounded-2xl bg-gray-100">
        <div className="flex items-center justify-center">
          <img className="w-full h-96" src="src/img" alt="img" />
        </div>
      </div>
      <div className="flex flex-col gap-2 w-80 px-4 py-4">
        <div className="flex flex-col mb-10  text-white items-end">
          <h1 className="text-2xl font-bold">
            0
          </h1>
          <p className="text-sm font-medium">
            - $0
            {' '}
            deposit upon any Vespa purchase
          </p>
        </div>
        <ul>
          <li className="py-2 px-2 flex justify-between text-sm bg-gray-300">
            <span>Finance fee</span>
            <span>
              $0
            </span>
          </li>
          <li className="py-2 px-2 flex justify-between text-white text-sm">
            <span>Option to purchase fee</span>
            <span>
              $0
            </span>
          </li>
          <li className="py-2 px-2 flex justify-between text-sm bg-gray-300">
            <span>Total amount of payable</span>
            <span>
              $0
            </span>
          </li>
          <li className="py-2 px-2 flex justify-between text-white text-sm">
            <span>Duration</span>
            <span>
              3
              {' '}
              month
            </span>
          </li>
        </ul>
        <div className="px-2 flex text-sm text-white font-medium">
          <span>
            <span className="font-extrabold">5.4% APR </span>
            Representative
          </span>
        </div>
        <div className="flex item-center justify-center pt-4">
          <a href="/reserve/1" className="px-12 pb-2 pt-2 rounded-lg text-white bg-customBg shadow-md hover:bg-customDark" type="button">Reserve</a>
        </div>
      </div>
    </div>
    <div className="absolute md:top-3/4 left-6 pb-4 mt-20">
      <a href="/motors" className="px-8 pb-2 pt-1 rounded-lg text-white bg-customBg shadow-md hover:bg-customDark" type="button">Back</a>
    </div>
  </div>
);
export default Details;
