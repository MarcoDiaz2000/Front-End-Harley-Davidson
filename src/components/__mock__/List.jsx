// Mock data for allReservations prop
export const allReservationsMock = [
  {
    id: 1,
    name: 'John Doe',
    username_id: 'johndoe123',
    item_id: 101,
    city: 'New York',
    date: '2023-10-15',
    removed: false,

  },
  {
    id: 2,
    name: 'Alice Smith',
    username_id: 'alicesmith456',
    item_id: 102,
    city: 'Los Angeles',
    date: '2023-10-20',
    removed: false,

  },
  // Add more reservation objects as needed
];

// Mock data for bikes prop
export const bikesMock = [
  {
    id: 101,
    name: 'Mountain Bike',
    description: 'A rugged mountain bike for outdoor adventures.',
    price: 599,
    image_url: '/images/mountain_bike.jpg',
    category: 'Mountain Bikes',
    city: 'New York',
  },
  {
    id: 102,
    name: 'Road Bike',
    description: 'A lightweight road bike for fast commuting.',
    price: 799,
    image_url: '/images/road_bike.jpg',
    category: 'Road Bikes',
    city: 'Los Angeles',
  },
  // Add more bike objects as needed
];
