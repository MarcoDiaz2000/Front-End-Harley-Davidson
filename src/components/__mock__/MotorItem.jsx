import React from 'react';

const MotorItem = () => (
  <div className="w-full h-96 card">
    <div className="w-full flex justify-center items-center h-64 bg-slate-500 p-4 m-2">
      <img className="w-80 h-auto" src="src/img" alt="motor" />
    </div>
    <h1 className="font-bold text-white bg-blur">Model 1</h1>
    <a
      href="/motors/1"
      className="px-16 pb-2 pt-1 bg-customBg flex justify-center text-white rounded-md text-center shadow-md hover:bg-customDark"
    >
      See Model
    </a>
  </div>
);
export default MotorItem;
