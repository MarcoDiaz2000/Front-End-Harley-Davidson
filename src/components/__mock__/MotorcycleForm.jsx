import React from 'react';

const MotorcycleForm = () => (
  <form>
    <div className="mb-3">
      <label htmlFor="name" className="text-white block mb-2 font-roboto font-bold">
        Name *:
        <input id="name" className="w-full p-2 rounded text-black" type="text" name="name" />
      </label>
    </div>
    <div className="mb-3">
      <label htmlFor="image" className="text-white block mb-2 font-roboto font-bold">
        Image *:
        <input
          id="image"
          className="w-full p-2 rounded text-black font-roboto font-bold"
          type="file"
          name="image"
        />
      </label>
    </div>
    <div className="mb-3">
      <label htmlFor="description" className="text-white block mb-2 font-roboto font-bold">
        Description *:
        <input
          id="description"
          className="w-full p-2 rounded text-black font-roboto font-bold"
          type="text"
          name="description"
        />
      </label>
    </div>
    <div className="mb-3">
      <label htmlFor="deposit" className="text-white block mb-2 font-roboto font-bold">
        Deposit:
        <input
          id="deposit"
          className="w-full p-2 rounded text-black"
          type="number"
          name="deposit"
        />
      </label>
    </div>
    <div className="mb-3">
      <label htmlFor="finance_fee" className="text-white block mb-2 font-roboto font-bold">
        Finance Fee:
        <input
          id="finance_fee"
          className="w-full p-2 rounded text-black"
          type="number"
          name="finance_fee"
        />
      </label>
    </div>
    <div className="mb-3">
      <label htmlFor="option_to_purchase_fee" className="text-white block mb-2 font-roboto font-bold">
        Option to Purchase Fee:
        <input
          id="option_to_purchase_fee"
          className="w-full p-2 rounded text-black"
          type="number"
          name="option_to_purchase_fee"
        />
      </label>
    </div>
    <div className="mb-3">
      <label htmlFor="total_amount_payable" className="text-white block mb-2 font-roboto font-bold">
        Total Amount Payable:
        <input
          id="total_amount_payable"
          className="w-full p-2 rounded text-black"
          type="number"
          name="total_amount_payable"
        />
      </label>
    </div>
    <div className="mb-3">
      <label htmlFor="duration" className="text-white block mb-2 font-roboto font-bold">
        Duration:
        <input
          id="duration"
          className="w-full p-2 rounded text-black"
          type="number"
          name="duration"
        />
      </label>
    </div>
    <div className="flex justify-center">
      <button
        type="submit"
        className="bg-harleyOrange text-white py-3 px-12 rounded hover:bg-orange-700 shadow-md font-roboto font-bold"
      >
        Add New Motorcycle
      </button>
    </div>
  </form>
);
export default MotorcycleForm;
