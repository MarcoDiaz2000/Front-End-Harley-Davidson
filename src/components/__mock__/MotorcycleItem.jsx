import React from 'react';

const MotorcycleItem = () => (
  <li className="flex flex-col sm:flex-row justify-between items-center p-2 my-2 rounded-lg bg-white white-shadow-i">
    <span className="font-roboto font-bold text-black">Motorcycle</span>
    <button
      type="button"
      className="p-2 rounded hover:bg-customDark-900 font-roboto font-bold text-white bg-harleyOrange"
    >
      Delete
    </button>
  </li>
);
export default MotorcycleItem;
