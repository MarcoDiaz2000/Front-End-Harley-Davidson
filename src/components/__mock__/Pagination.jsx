// Mock data for Pagination component
export const currentPageMock = 1;
export const totalPagesMock = 3;

// Mock function for onPageChange
export const onPageChangeMock = jest.fn();
