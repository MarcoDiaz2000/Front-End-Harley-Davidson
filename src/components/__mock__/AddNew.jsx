import React, { useState } from 'react';
import MotorcycleForm from './MotorcycleForm';

const MockAddNew = () => {
  const [newMotor, setNewMotor] = useState({
    name: '',
    description: '',
    deposit: 0,
    finance_fee: 0,
    option_to_purchase_fee: 0,
    total_amount_payable: 0,
    duration: 0,
  });

  const handleChange = (e) => {
    const { name, value } = e.target;
    setNewMotor({
      ...newMotor,
      [name]: value,
    });
  };

  const handleSubmit = (event) => {
    event.preventDefault();
  };

  return (
    <div className="flex justify-center bodyBg overflow-y-auto pt-12 pb-12 min-h-screen">
      <div className="bg-customBg p-5 rounded-lg w-full sm:w-1/2 mb-12 white-shadow flex-shrink-0">
        <h1 className="text-white text-3xl mb-4 font-roboto font-bold">Add New Motorcycle</h1>
        <MotorcycleForm
          newMotor={newMotor}
          handleChange={handleChange}
          handleSubmit={handleSubmit}
        />
      </div>
    </div>
  );
};
export default MockAddNew;
