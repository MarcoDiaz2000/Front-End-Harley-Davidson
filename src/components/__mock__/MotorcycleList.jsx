import React from 'react';
import PropTypes from 'prop-types';
import MotorcycleItem from './MotorcycleItem';

const MotorcycleList = ({ bikes }) => (
  <ul className="divide-customBorder">
    {bikes.map((motorcycle) => (
      <MotorcycleItem key={motorcycle.id} motorcycle={motorcycle} />
    ))}
  </ul>
);

MotorcycleList.propTypes = {
  bikes: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      removed: PropTypes.bool.isRequired,
    }),
  ).isRequired,
};
export default MotorcycleList;
