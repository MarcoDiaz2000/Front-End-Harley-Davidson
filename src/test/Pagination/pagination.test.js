import React from 'react';
import renderer from 'react-test-renderer';
import Pagination from '../../components/myReservations/pagination';
import { currentPageMock, totalPagesMock, onPageChangeMock } from '../../components/__mock__/Pagination';

describe('Pagination Snapshot', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <Pagination
        currentPage={currentPageMock}
        totalPages={totalPagesMock}
        onPageChange={onPageChangeMock}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
