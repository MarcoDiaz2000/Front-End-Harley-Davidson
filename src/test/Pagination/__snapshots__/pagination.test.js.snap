// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Snapshot matches snapshot 1`] = `
<div
  className="flex justify-center my-4"
>
  <button
    className="mx-2 px-4 py-2 border text-white bg-customBg text-white"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="mx-2 px-4 py-2 border text-white border-gray-300 hover:bg-customBg hover:text-white"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="mx-2 px-4 py-2 border text-white border-gray-300 hover:bg-customBg hover:text-white"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
</div>
`;
